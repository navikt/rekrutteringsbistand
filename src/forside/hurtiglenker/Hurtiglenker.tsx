import { BodyShort } from '@navikt/ds-react';
import classNames from 'classnames';
import { ReactComponent as FinnKandidaterIkon } from 'felles/komponenter/piktogrammer/finn-kandidater.svg';
import { ReactComponent as FinnStillinger } from 'felles/komponenter/piktogrammer/finn-stillinger.svg';
import { ReactComponent as OpprettNyStillingIkon } from 'felles/komponenter/piktogrammer/opprett-ny-stilling.svg';
import { ReactComponent as SeMineStillingerIkon } from 'felles/komponenter/piktogrammer/se-mine-stillinger.svg';
import { FunctionComponent, ReactNode } from 'react';
import { Link } from 'react-router-dom';
import css from './Hurtiglenker.module.css';

const Hurtiglenker: FunctionComponent = () => {
    return (
        <nav className={css.hurtiglenker}>
            <LenkepanelMedIkon
                href="/kandidatsok"
                tittel="Finn kandidater"
                ikon={<FinnKandidaterIkon />}
            />
            <LenkepanelMedIkon
                href="/stillingssok?brukStandardsok=true"
                tittel="Finn stillinger"
                ikon={<FinnStillinger />}
            />
            <LenkepanelMedIkon
                href="/stillinger/minestillinger"
                tittel="Se mine stillinger"
                ikon={<SeMineStillingerIkon />}
            />
            <LenkepanelMedIkon
                href="/stillinger/minestillinger?visOpprettStillingModal"
                tittel="Opprett ny stilling"
                ikon={<OpprettNyStillingIkon />}
            />
        </nav>
    );
};

const LenkepanelMedIkon: FunctionComponent<{
    tittel: string;
    href: string;
    ikon: ReactNode;
}> = ({ tittel, href, ikon }) => (
    <Link to={href} className={classNames('navds-link-panel', css.lenkepanel)}>
        <div className={css.lenkeinnhold}>
            {ikon}
            <BodyShort as="span">{tittel}</BodyShort>
        </div>
    </Link>
);

export default Hurtiglenker;
