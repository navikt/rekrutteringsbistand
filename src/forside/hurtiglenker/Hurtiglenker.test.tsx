import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Hurtiglenker from './Hurtiglenker';

test('<Hurtiglenker/>', () => {
    const app = render(<Hurtiglenker />, { wrapper: BrowserRouter });
    expect(app.getAllByRole('link')).toHaveLength(4);
    expect(app).toMatchSnapshot();
});
