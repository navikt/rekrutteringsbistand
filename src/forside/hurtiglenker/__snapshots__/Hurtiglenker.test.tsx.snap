// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Hurtiglenker/> 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="_hurtiglenker_54ea4a"
      >
        <a
          class="navds-link-panel _lenkepanel_54ea4a"
          href="/kandidatsok"
        >
          <div
            class="_lenkeinnhold_54ea4a"
          >
            <svg
              fill="none"
              height="64"
              width="64"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#a)"
              >
                <path
                  d="M50.67 32A18.67 18.67 0 1 0 32 50.67"
                  stroke="#E0D8E9"
                  stroke-linecap="round"
                  stroke-width="8"
                />
                <circle
                  cx="57.33"
                  cy="6.67"
                  r="5.67"
                  stroke="#262626"
                  stroke-width="2"
                />
                <circle
                  cx="1.33"
                  cy="1.33"
                  r="1.33"
                  stroke="#262626"
                  stroke-width="2"
                  transform="matrix(1 0 0 -1 46.67 5.33)"
                />
                <circle
                  cx="38.67"
                  cy="6.67"
                  r="5.67"
                  stroke="#262626"
                  stroke-width="2"
                />
                <path
                  d="M43.58 5.33a7.96 7.96 0 0 1 8.84 0M34.32 63.9c-1.97-7.85-8.9-13.64-17.16-13.64S1.96 56.06 0 63.9M27.98 34.93c0 6.03-4.92 10.92-11.01 10.92S5.96 40.95 5.96 34.93C5.96 28.9 10.88 24 16.97 24s11.01 4.9 11.01 10.93Z"
                  stroke="#262626"
                  stroke-width="2"
                />
              </g>
              <defs>
                <clippath
                  id="a"
                >
                  <path
                    d="M0 0h64v64H0z"
                    fill="#fff"
                  />
                </clippath>
              </defs>
            </svg>
            <span
              class="navds-body-short"
            >
              Finn kandidater
            </span>
          </div>
        </a>
        <a
          class="navds-link-panel _lenkepanel_54ea4a"
          href="/stillingssok?brukStandardsok=true"
        >
          <div
            class="_lenkeinnhold_54ea4a"
          >
            <svg
              fill="none"
              height="64"
              width="64"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#a)"
              >
                <path
                  d="M36.89 21.9 56 41.02 41.35 55.66 22.24 36.55l-8.31 8.31L8 7.66l37.2 5.93-8.31 8.31Z"
                  fill="#E0D8E9"
                />
                <path
                  d="M64 63.9c-1.87-7.46-8.47-12.97-16.32-12.97s-14.45 5.5-16.32 12.96M57.92 36.34c0 5.71-4.66 10.35-10.42 10.35a10.38 10.38 0 0 1-10.43-10.35c0-5.7 4.66-10.34 10.43-10.34 5.76 0 10.42 4.64 10.42 10.34ZM19.33 19.66V20c0-3.3-2.09-5.33-4.66-5.33C12.09 14.67 10 16.7 10 20v-.34"
                  stroke="#262626"
                  stroke-width="2"
                />
                <circle
                  cx="14.67"
                  cy="10"
                  r="3.67"
                  stroke="#262626"
                  stroke-width="2"
                />
                <path
                  d="M1 1h27.33v37.33H1zM6.67 24.67H18M6.67 31.33h5.66"
                  stroke="#262626"
                  stroke-width="2"
                />
              </g>
              <defs>
                <clippath
                  id="a"
                >
                  <path
                    d="M0 0h64v64H0z"
                    fill="#fff"
                  />
                </clippath>
              </defs>
            </svg>
            <span
              class="navds-body-short"
            >
              Finn stillinger
            </span>
          </div>
        </a>
        <a
          class="navds-link-panel _lenkepanel_54ea4a"
          href="/stillinger/minestillinger"
        >
          <div
            class="_lenkeinnhold_54ea4a"
          >
            <svg
              fill="none"
              height="64"
              width="64"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#a)"
              >
                <path
                  d="M12.43 35.82 32 55.33l19.58-19.51a11.86 11.86 0 0 0-16.75-16.8L32 21.83l-2.82-2.81a11.86 11.86 0 0 0-16.75 16.8Z"
                  fill="#CCF1D6"
                />
                <path
                  d="m43.77 11.23 15 15.15c.99 1 .99 2.63 0 3.64-1 1-2.61 1-3.6 0l-5.4-5.46 4.2 4.24c.99 1 .99 2.64 0 3.64-1 1-2.6 1-3.6 0l-4.2-4.24 3 3.03c.99 1 .99 2.63 0 3.63-1 1-2.6 1-3.6 0l-3-3.03 1.76 1.79c1.01 1.02 1 2.67-.03 3.66-1 .97-2.57.96-3.55-.02L26 22.56V8.84h10.18m8.79 1.18-5.4 5.45a3.37 3.37 0 0 1-4.8 0l-1.22-1.22a1.73 1.73 0 0 1-.01-2.41l8.52-8.8a3.37 3.37 0 0 1 4.73-.13l6.36 5.93h10.18v13.9h-8.16M1 40.67c0-.19.15-.34.33-.34H36c.18 0 .33.15.33.34v10a1 1 0 0 1-1 1H2a1 1 0 0 1-1-1v-10ZM3 62V51.67h31.33V62a1 1 0 0 1-1 1H4a1 1 0 0 1-1-1ZM13.67 36c0-.92.74-1.67 1.66-1.67H22c.92 0 1.67.75 1.67 1.67v4.33h-10V36Z"
                  stroke="#262626"
                  stroke-width="2"
                />
                <path
                  d="M18.67 48.67V54"
                  stroke="#262626"
                  stroke-linecap="round"
                  stroke-width="2"
                />
              </g>
              <defs>
                <clippath
                  id="a"
                >
                  <path
                    d="M0 0h64v64H0z"
                    fill="#fff"
                  />
                </clippath>
              </defs>
            </svg>
            <span
              class="navds-body-short"
            >
              Se mine stillinger
            </span>
          </div>
        </a>
        <a
          class="navds-link-panel _lenkepanel_54ea4a"
          href="/stillinger/minestillinger?visOpprettStillingModal"
        >
          <div
            class="_lenkeinnhold_54ea4a"
          >
            <svg
              fill="none"
              height="64"
              width="64"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M39 39.33H20.33M51 30.67H15M45.67 21.33H25"
                stroke="#262626"
                stroke-width="2"
              />
              <path
                d="M19.33 15.33h23.33v4H19.33zM24 39.33h18.67V46H24z"
                fill="#CCF1D6"
              />
              <circle
                cx="32"
                cy="31"
                fill="#CCF1D6"
                r="26"
              />
              <path
                d="M44.33 31 28.61 47.33 23 41.7 33.1 31 23 20.3l5.61-5.63L44.33 31Z"
                stroke="#262626"
                stroke-width="2"
              />
            </svg>
            <span
              class="navds-body-short"
            >
              Opprett ny stilling
            </span>
          </div>
        </a>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="_hurtiglenker_54ea4a"
    >
      <a
        class="navds-link-panel _lenkepanel_54ea4a"
        href="/kandidatsok"
      >
        <div
          class="_lenkeinnhold_54ea4a"
        >
          <svg
            fill="none"
            height="64"
            width="64"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#a)"
            >
              <path
                d="M50.67 32A18.67 18.67 0 1 0 32 50.67"
                stroke="#E0D8E9"
                stroke-linecap="round"
                stroke-width="8"
              />
              <circle
                cx="57.33"
                cy="6.67"
                r="5.67"
                stroke="#262626"
                stroke-width="2"
              />
              <circle
                cx="1.33"
                cy="1.33"
                r="1.33"
                stroke="#262626"
                stroke-width="2"
                transform="matrix(1 0 0 -1 46.67 5.33)"
              />
              <circle
                cx="38.67"
                cy="6.67"
                r="5.67"
                stroke="#262626"
                stroke-width="2"
              />
              <path
                d="M43.58 5.33a7.96 7.96 0 0 1 8.84 0M34.32 63.9c-1.97-7.85-8.9-13.64-17.16-13.64S1.96 56.06 0 63.9M27.98 34.93c0 6.03-4.92 10.92-11.01 10.92S5.96 40.95 5.96 34.93C5.96 28.9 10.88 24 16.97 24s11.01 4.9 11.01 10.93Z"
                stroke="#262626"
                stroke-width="2"
              />
            </g>
            <defs>
              <clippath
                id="a"
              >
                <path
                  d="M0 0h64v64H0z"
                  fill="#fff"
                />
              </clippath>
            </defs>
          </svg>
          <span
            class="navds-body-short"
          >
            Finn kandidater
          </span>
        </div>
      </a>
      <a
        class="navds-link-panel _lenkepanel_54ea4a"
        href="/stillingssok?brukStandardsok=true"
      >
        <div
          class="_lenkeinnhold_54ea4a"
        >
          <svg
            fill="none"
            height="64"
            width="64"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#a)"
            >
              <path
                d="M36.89 21.9 56 41.02 41.35 55.66 22.24 36.55l-8.31 8.31L8 7.66l37.2 5.93-8.31 8.31Z"
                fill="#E0D8E9"
              />
              <path
                d="M64 63.9c-1.87-7.46-8.47-12.97-16.32-12.97s-14.45 5.5-16.32 12.96M57.92 36.34c0 5.71-4.66 10.35-10.42 10.35a10.38 10.38 0 0 1-10.43-10.35c0-5.7 4.66-10.34 10.43-10.34 5.76 0 10.42 4.64 10.42 10.34ZM19.33 19.66V20c0-3.3-2.09-5.33-4.66-5.33C12.09 14.67 10 16.7 10 20v-.34"
                stroke="#262626"
                stroke-width="2"
              />
              <circle
                cx="14.67"
                cy="10"
                r="3.67"
                stroke="#262626"
                stroke-width="2"
              />
              <path
                d="M1 1h27.33v37.33H1zM6.67 24.67H18M6.67 31.33h5.66"
                stroke="#262626"
                stroke-width="2"
              />
            </g>
            <defs>
              <clippath
                id="a"
              >
                <path
                  d="M0 0h64v64H0z"
                  fill="#fff"
                />
              </clippath>
            </defs>
          </svg>
          <span
            class="navds-body-short"
          >
            Finn stillinger
          </span>
        </div>
      </a>
      <a
        class="navds-link-panel _lenkepanel_54ea4a"
        href="/stillinger/minestillinger"
      >
        <div
          class="_lenkeinnhold_54ea4a"
        >
          <svg
            fill="none"
            height="64"
            width="64"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#a)"
            >
              <path
                d="M12.43 35.82 32 55.33l19.58-19.51a11.86 11.86 0 0 0-16.75-16.8L32 21.83l-2.82-2.81a11.86 11.86 0 0 0-16.75 16.8Z"
                fill="#CCF1D6"
              />
              <path
                d="m43.77 11.23 15 15.15c.99 1 .99 2.63 0 3.64-1 1-2.61 1-3.6 0l-5.4-5.46 4.2 4.24c.99 1 .99 2.64 0 3.64-1 1-2.6 1-3.6 0l-4.2-4.24 3 3.03c.99 1 .99 2.63 0 3.63-1 1-2.6 1-3.6 0l-3-3.03 1.76 1.79c1.01 1.02 1 2.67-.03 3.66-1 .97-2.57.96-3.55-.02L26 22.56V8.84h10.18m8.79 1.18-5.4 5.45a3.37 3.37 0 0 1-4.8 0l-1.22-1.22a1.73 1.73 0 0 1-.01-2.41l8.52-8.8a3.37 3.37 0 0 1 4.73-.13l6.36 5.93h10.18v13.9h-8.16M1 40.67c0-.19.15-.34.33-.34H36c.18 0 .33.15.33.34v10a1 1 0 0 1-1 1H2a1 1 0 0 1-1-1v-10ZM3 62V51.67h31.33V62a1 1 0 0 1-1 1H4a1 1 0 0 1-1-1ZM13.67 36c0-.92.74-1.67 1.66-1.67H22c.92 0 1.67.75 1.67 1.67v4.33h-10V36Z"
                stroke="#262626"
                stroke-width="2"
              />
              <path
                d="M18.67 48.67V54"
                stroke="#262626"
                stroke-linecap="round"
                stroke-width="2"
              />
            </g>
            <defs>
              <clippath
                id="a"
              >
                <path
                  d="M0 0h64v64H0z"
                  fill="#fff"
                />
              </clippath>
            </defs>
          </svg>
          <span
            class="navds-body-short"
          >
            Se mine stillinger
          </span>
        </div>
      </a>
      <a
        class="navds-link-panel _lenkepanel_54ea4a"
        href="/stillinger/minestillinger?visOpprettStillingModal"
      >
        <div
          class="_lenkeinnhold_54ea4a"
        >
          <svg
            fill="none"
            height="64"
            width="64"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M39 39.33H20.33M51 30.67H15M45.67 21.33H25"
              stroke="#262626"
              stroke-width="2"
            />
            <path
              d="M19.33 15.33h23.33v4H19.33zM24 39.33h18.67V46H24z"
              fill="#CCF1D6"
            />
            <circle
              cx="32"
              cy="31"
              fill="#CCF1D6"
              r="26"
            />
            <path
              d="M44.33 31 28.61 47.33 23 41.7 33.1 31 23 20.3l5.61-5.63L44.33 31Z"
              stroke="#262626"
              stroke-width="2"
            />
          </svg>
          <span
            class="navds-body-short"
          >
            Opprett ny stilling
          </span>
        </div>
      </a>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
